docs/develop/cmodules.rst

```diff
@@ -0,0 +1,86 @@
+Extending MicroPython with C
+============================
+
+Some specialized code would be unacceptably slow or needs to access hardware in
```

Comment:
Somehow this intro paragraph sounds weird to me. I'd take pass to clean up grammar/wording, and maybe clarify it. E.g. if you want discourage people from jumping on the C side too quick, makes sense to do that more actively than just "please take a look". Alternatively, may assume that people who arrive here already know what they need, so can remove that altogether.